rig/machine_control/machine_controller.py

```diff
@@ -1427,6 +1460,22 @@ def __init__(self, machine_controller, x, y, start_address, end_address):
         # Current offset from start address
         self._offset = 0
 
+    @property
+    def buffer_size(self):
+        """Return the number of bytes in the write buffer."""
+        return self._write_buffer.buffer_size
+
+    def close(self):
+        """Flush and close the file-like."""
+        if not self._closed:
+            self.flush()
+            self._closed = True
+
+    @property
```

Comment:
Yes, that makes more sense - I would personally make that a property too to
make it's readonlyness explicit, but I realise that possibly betrays my C++
fear of public members variables...

On 1 May 2015 at 09:22, Jonathan Heathcote <notifications@github.com> wrote:

> In rig/machine_control/machine_controller.py
> <https://github.com/project-rig/rig/pull/125#discussion_r29493557>:
>
> > @@ -1427,6 +1460,22 @@ def __init__(self, machine_controller, x, y, start_address, end_address):
> >          # Current offset from start address
> >          self._offset = 0
> >
> > +    @property
> > +    def buffer_size(self):
> > +        """Return the number of bytes in the write buffer."""
> > +        return self._write_buffer.buffer_size
> > +
> > +    def close(self):
> > +        """Flush and close the file-like."""
> > +        if not self._closed:
> > +            self.flush()
> > +            self._closed = True
> > +
> > +    @property
>
> Aha! @mundya <https://github.com/mundya> has just pointed out to me that
> we might be talking about different things: I was talking about the closed
> property. I presume you are talking about the buffer_size property? The
> GitHub line comment stuff is displayed a bit confusingly in summaries...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/project-rig/rig/pull/125/files#r29493557>.
>